web/services/AI/ai_model.py

```
@@ -0,0 +1,177 @@
+
+        # Try to extract JSON from the response
+        json_match = re.search(r"\{.*?\}", response.text, re.DOTALL)  # non-greedy
+        if json_match:
+            json_str = json_match.group(0)
+            response_object = json.loads(json_str)
+        else:
+            raise ValueError("No JSON object found in response")
+
+        return response_object
+
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)